speech-commands/demo/index.js

```diff
@@ -324,3 +379,99 @@ startTransferLearnButton.addEventListener('click', async () => {
   startTransferLearnButton.textContent = 'Transfer learning complete.';
   startButton.disabled = false;
 });
+
+downloadFilesButton.addEventListener('click', () => {
+  const basename = getDatasetFileBasename();
+  const artifacts = transferRecognizer.serializeExamples();
+
+  // Trigger downloading of the manifest .json file.
+  const manifestAnchor = document.createElement('a');
+  manifestAnchor.download = `${basename}.json`;
+  manifestAnchor.href = window.URL.createObjectURL(
+      new Blob([artifacts.manifest], {type: 'application/json'}));
+  manifestAnchor.click();
+
+  // Trigger downloading of the data .bin file.
+  const weightDataAnchor = document.createElement('a');
+  weightDataAnchor.download = `${basename}.bin`;
+  weightDataAnchor.href = window.URL.createObjectURL(
+      new Blob([artifacts.data], {type: 'application/octet-stream'}));
+  weightDataAnchor.click();
+});
+
+/** Get the base name of the downloaded files based on current dataset. */
+function getDatasetFileBasename() {
+  if (transferRecognizer == null) {
+    throw new Error('Transfer model is unset.');
+  }
+  const items = [];
+  const exampleCounts = transferRecognizer.countExamples();
+  const labels = Object.keys(exampleCounts);
+  labels.sort();
+  for (const label of labels) {
+    const count = exampleCounts[label];
+    items.push(`${count}${label}`);
+  }
+  return items.join('-');
+}
+
+uploadFilesButton.addEventListener('click', async () => {
+  const files = datasetFilesInput.files;
+  if (files.length !== 2) {
+    throw new Error('Must select exactly two files.');
+  }
+
+  let manifestFile;
+  let dataFile;
+  if (files[0].name.endsWith('.json') && files[1].name.endsWith('.bin')) {
+    manifestFile = files[0];
+    dataFile = files[1];
+  } else if (files[0].name.endsWith('.bin') && files[1].name.endsWith('.json')) {
+    manifestFile = files[1];
+    dataFile = files[0];
+  } else {
+    throw new Error('Must select a .json file and a .bin file');
+  }
+
+  const manifestReader = new FileReader();
```

Comment:
wow neat! I didn't know how to do any of this stuff. Your bag of browser tricks is top notch :)